app/src/androidTest/java/me/damonkelley/tictactoe_app/BoardAdapterTest.java

```diff
@@ -0,0 +1,78 @@
+package me.damonkelley.tictactoe_app;
+
+import android.support.test.InstrumentationRegistry;
+import android.support.test.filters.SmallTest;
+import android.support.test.runner.AndroidJUnit4;
+import android.test.mock.MockContext;
+import android.widget.TextView;
+import me.damonkelley.tictactoe.Board;
+import me.damonkelley.tictactoe.Marker;
+import me.damonkelley.tictactoe.Space;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static junit.framework.Assert.assertEquals;
+
+@RunWith(AndroidJUnit4.class)
+@SmallTest
+public class BoardAdapterTest {
+
+    private Board board;
+    private BoardAdapter adapter;
+
+    @Before
+    public void setUp() throws Exception {
+        board = new Board();
+        adapter = new BoardAdapter(InstrumentationRegistry.getContext(), board);
+    }
+
+    @Test
+    public void itGetsAViewAtThePosition() {
+        board.put(new Space(0, 0), Marker.X);
+
+        assertEquals(Marker.X, adapter.getItem(0));
+        assertEquals(null, adapter.getItem(1));
+    }
+
+    @Test
+    public void itGetsAViewAtThePositionForA4By4Board() {
+        Board board = new Board(4);
+        BoardAdapter adapter = new BoardAdapter(new MockContext(), board);
+
+        board.put(new Space(3, 3), Marker.X);
+
+        assertEquals(Marker.X, adapter.getItem(15));
+        assertEquals(null, adapter.getItem(1));
+    }
+
+    @Test
+    public void theCount() {
+        assertEquals(9, new BoardAdapter(new MockContext(), new Board()).getCount());
+        assertEquals(16, new BoardAdapter(new MockContext(), new Board(4)).getCount());
+    }
+
+    @Test
+    public void allItemIdsAreZero() {
+        assertEquals(0, adapter.getItemId(1));
+        assertEquals(0, adapter.getItemId(2));
+        assertEquals(0, adapter.getItemId(3));
+    }
+
+    @Test
+    @Ignore(value = "Unable to make the Resource available for inflation during testing")
```

Comment:
I was unable to get the test environment to find the `R.layout.view_space`, which is being in inflated in this method. I spent a couple hours experimenting and looking for a solution and came up with nothing. It is still informative, so I decided to ignore it rather than delete it.

This method is being covered in the functional test in `UITest`. The `BoardAdapter` works fine in the test environment when it is instantiated inside of the `GameActivity`